guides/testing.md

```diff
@@ -0,0 +1,106 @@
+# Testing Crystal Code
+
+Crystal comes with a full-feature spec library to write test cases in the [`Spec`module](https://crystal-lang.org/api/latest/Spec.html).
+
+It is inspired by [Rspec](https://relishapp.com/rspec).
+
+## Anatomy of a spec file
+
+To use the spec module and DSL, you need to add `require "spec"` to your spec files. Many projects use a custom [spec helper](#spec-helper) which organizes these includes.
+
+Test files are structured through `describe` and `context` sections. Typically a top level `describe` defines the outer unit (such as a class) to be tested by the spec. Further `describe` sections can be nested within the outer unit to specify smaller units under test (such as individual methods).
```

Comment:
Perhaps note that these sections are only for humans, and are used only in the spec output and when reading the spec source? I'm not sure if it's worth saying.

Reply:
I'm not sure about the benefit of elaborating this.
It's also relevant for running specs, you can select only a specific describe/context section to be run.